cmd/agent/app/check.go

```diff
@@ -127,19 +129,47 @@ var checkCmd = &cobra.Command{
 			fmt.Println("Multiple check instances found, running each of them")
 		}
 
+		var instancesData []interface{}
+
 		for _, c := range cs {
 			s := runCheck(c, agg)
 
 			// Sleep for a while to allow the aggregator to finish ingesting all the metrics/events/sc
 			time.Sleep(time.Duration(checkDelay) * time.Millisecond)
 
-			printMetrics(agg)
+			if formatJSON {
+				aggregatorData := getMetricsData(agg)
+				var collectorData map[string]interface{}
+
+				collectorJSON, _ := status.GetCheckStatusJSON(c, s)
+				json.Unmarshal(collectorJSON, &collectorData)
 
-			checkStatus, _ := status.GetCheckStatus(c, s)
-			fmt.Println(string(checkStatus))
+				checkRuns := collectorData["runnerStats"].(map[string]interface{})["Checks"].(map[string]interface{})[checkName].(map[string]interface{})
```

Comment:
Same goes for those unchecked type asserts. At some point I'm wondering if we shouldn't create a special `struct` to Unmarshal this specific data? By doing this you would manipulate a cleaner object and you would only have to check the Unmarshal err to ensure you have the data you want.